blockchain/vm/internaltx_trace_json_test.go

```diff
@@ -0,0 +1,263 @@
+    "gas": "0x5e106",
+    "gasUsed": "0x5e106",
+    "input": "0x6080604052348015600f57600080fd5b50604051602080608183398101806040526020811015602d57600080fd5b810190808051906020019092919050505050603580604c6000396000f3fe6080604052600080fdfea165627a7a72305820c577c5325e4c667e8439f998d616e74110a14fb5b185afe46d8443bbfd9c3d710029",
+    "output": "0x6080604052600080fdfea165627a7a72305820c577c5325e4c667e8439f998d616e74110a14fb5b185afe46d8443bbfd9c3d710029"
+}`)),
+		},
+		{
+			name: "call_test",
+			fields: fields{
+				Type:    CALL.String(),
+				From:    genAddressPtr("0x3693da93b9d5e63cb4804b8813c8214a76724485"),
+				To:      genAddressPtr("0x13cf0d18bcb898efd4e85ae2bb65a443ab86023c"),
+				Value:   "0x0",
+				Gas:     179,
+				GasUsed: 122,
+				Input:   "0xc605f76c",
+				Output:  "0x",
+			},
+			want: jsonMustCompact([]byte(`{
+	"type": "CALL",
+	"from": "0x3693da93b9d5e63cb4804b8813c8214a76724485",
+	"to": "0x13cf0d18bcb898efd4e85ae2bb65a443ab86023c",
+	"value": "0x0",
+	"gas": "0xb3",
+	"gasUsed": "0x7a",
+	"input": "0xc605f76c",
+	"output": "0x"
+}`)),
+		},
+		{
+			name: "simple_call_test",
+			fields: fields{
+				Type:  CALL.String(),
+				From:  genAddressPtr("0x3b873a919aa0512d5a0f09e6dcceaa4a6727fafe"),
+				To:    genAddressPtr("0x0024f658a46fbb89d8ac105e98d7ac7cbbaf27c5"),
+				Value: "0x6f05b59d3b20000",
+				Input: "0x",
+			},
+			want: jsonMustCompact([]byte(`{
+		"type": "CALL",
+        "from": "0x3b873a919aa0512d5a0f09e6dcceaa4a6727fafe",
+		"to": "0x0024f658a46fbb89d8ac105e98d7ac7cbbaf27c5",
+		"value": "0x6f05b59d3b20000",
+        "input": "0x"
+}`)),
+		},
+		{
+			name: "staticcall_test",
+			fields: fields{
+				Type:    OpCode(STATICCALL).String(),
+				From:    genAddressPtr("0xbadc777c579b497ede07fa6ff93bdf4e31793f24"),
+				To:      genAddressPtr("0x920fd5070602feaea2e251e9e7238b6c376bcae5"),
+				Gas:     179,
+				GasUsed: 122,
+				Input:   "0xc605f76c",
+				Output:  "0x",
+			},
+			want: jsonMustCompact([]byte(`{
+		"type": "STATICCALL",
+        "from": "0xbadc777c579b497ede07fa6ff93bdf4e31793f24",
+		"to": "0x920fd5070602feaea2e251e9e7238b6c376bcae5",
+        "gas": "0xb3",
+        "gasUsed": "0x7a",
+        "input": "0xc605f76c",
+        "output": "0x"
+}`)),
+		},
+		{
+			name: "selfdestruct_test",
+			fields: fields{
+				Type: OpCode(SELFDESTRUCT).String(),
+			},
+			want:    []byte(`{"type":"SELFDESTRUCT"}`),
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			i := InternalTxTrace{
+				Type:     tt.fields.Type,
+				From:     tt.fields.From,
+				To:       tt.fields.To,
+				Value:    tt.fields.Value,
+				Gas:      tt.fields.Gas,
+				GasUsed:  tt.fields.GasUsed,
+				Input:    tt.fields.Input,
+				Output:   tt.fields.Output,
+				Error:    tt.fields.Error,
+				Time:     tt.fields.Time,
+				Calls:    tt.fields.Calls,
+				Reverted: tt.fields.Reverted,
+			}
+			got, err := i.MarshalJSON()
+			if (err != nil) != tt.wantErr {
```

Comment:
```suggestion
			if (err != nil) && (err != tt.wantErr) {
```

Reply:
I removed checking returned error.
(There is no TC for returning an error. I can't find any case to return a json marshalling internal error.)